### PR TITLE
feat: init react with vite, adding side bar and router

### PR DESCRIPTION
**Pull request by almog8k**
Setting up react app with vite.
Adding React router dom for navigation between pages.
Add Layout component with side bar

### CODE REVIEW

File: .licrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+[dependencies]
+# This will allow users to flag some dependencies so that Licensebat will not check for their license.
+ignored=["ignored_dep1", "ignored_dep2"]
+
+[behavior]
+# False by default, if true, it will only run the checks when one of the dependency files or the .licrc file has been modified.
+run_only_on_dependency_modification = true
+# False by default, if true, it will never block the build.
+do_not_block_pr = true
```
Comment (CptSchnitz) [inline .licrc:19]: EOL :))))))

File: src/layout/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      onClick: () => navigate(routes.CONFIG),
+    },
+    {
+      text: "Schema",
+      icon: <SchemaIcon />,
+      onClick: () => navigate(routes.SCHEMA),
+    },
+  ];
+  return (
+    <div>
```
Comment (CptSchnitz) [inline src/layout/layout.tsx:43]: In the persistent drawer example they used the `Box` component, we can use it as well.

File: src/layout/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      text: "Config",
+      icon: <ConfigIcon />,
+      onClick: () => navigate(routes.CONFIG),
+    },
+    {
+      text: "Schema",
+      icon: <SchemaIcon />,
+      onClick: () => navigate(routes.SCHEMA),
+    },
+  ];
```
Comment (CptSchnitz) [inline src/layout/layout.tsx:41]: this can be outside the function component as its static.

File: src/pages/configPages/configs/configsPage.tsx
Comment (CptSchnitz) [inline src/pages/configPages/configs/configsPage.tsx:1]: Please explain the idea behind the folder structure for this file, i think its too nested.

File: src/pages/configPages/cretaeConfig/createConfig.tsx
Comment (CptSchnitz) [inline src/pages/configPages/cretaeConfig/createConfig.tsx:1]: Fix typo in folder name

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
GJ!
Please add the following stuff from the boilerplate - prettier (with mapcolonies config), husky, commitlint, pretty-quick.

File: src/pages/configPages/configs/configsPage.tsx
Comment (almog8k) [reply src/pages/configPages/configs/configsPage.tsx:1]: took it out one level

File: src/layout/layout.tsx (hunk 2 above)
Comment (almog8k) [reply src/layout/layout.tsx:43]: excellent idea!

File: package copy.json
Comment (CptSchnitz) [inline package copy.json:1]: delete this file

File: package.json [hunk 4]
```diff
@@ -7,19 +7,34 @@
...
+    "format:fix": "prettier --write .",
+    "prepare": "husky"
   },
+
+  "husky": {
+    "hooks": {
+      "pre-commit": "pretty-quick --staged",
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
+    }
+  },
```
Comment (CptSchnitz) [inline package.json:21]: this is not needed anymore

File: src/routing/router.tsx [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
+import { NotFoundPage } from '../pages/notFound/notFound';
 
 export const Router: React.FC = () => {
   const Routes: React.FC = () => {
     return useRoutes([
-      { path: "*", element: <Navigate to={routes.NOT_FOUND} /> },
+      { path: '*', element: <Navigate to={routes.NOT_FOUND} /> },
       { path: routes.CONFIG, element: <ConfigsPage /> },
       { path: routes.SCHEMA, element: <SchemasPage /> },
+      { path: routes.NOT_FOUND, element: <NotFoundPage /> },
```
Comment (CptSchnitz) [inline src/routing/router.tsx:15]: you are missing HOME path, right now it redirects to 404.
You can send it to the config page for now.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Pull request merged**